From: Jason Baron <jbaron@redhat.com>
Date: Thu, 26 Mar 2009 22:10:47 -0400
Subject: [trace] change rcu_read_sched -> rcu_read
Message-id: 200903270210.n2R2AliK030362@ns3.rdu.redhat.com
O-Subject: [RHEL 5.4 PATCH 09/10] - change rcu_read_sched -> rcu_read
Bugzilla: 465543

Since we don't compile with preempt change rcu_read_sched to rcu_read, which
is what we have in RHEL5.

diff --git a/kernel/tracepoint.c b/kernel/tracepoint.c
index e96590f..3176e32 100644
--- a/kernel/tracepoint.c
+++ b/kernel/tracepoint.c
@@ -83,7 +83,7 @@ static inline void release_probes(void *old)
 	if (old) {
 		struct tp_probes *tp_probes = container_of(old,
 			struct tp_probes, probes[0]);
-		call_rcu_sched(&tp_probes->u.rcu, rcu_free_old_probes);
+		call_rcu(&tp_probes->u.rcu, rcu_free_old_probes);
 	}
 }
 
@@ -465,7 +465,7 @@ void tracepoint_probe_update_all(void)
 	tracepoint_update_probes();
 	list_for_each_entry_safe(pos, next, &release_probes, u.list) {
 		list_del(&pos->u.list);
-		call_rcu_sched(&pos->u.rcu, rcu_free_old_probes);
+		call_rcu(&pos->u.rcu, rcu_free_old_probes);
 	}
 }
 EXPORT_SYMBOL_GPL(tracepoint_probe_update_all);
